Gems/Archive/Code/Source/Clients/ArchiveReader.h

```diff
@@ -0,0 +1,217 @@
+        //! The outputSpan should be a pre-allocated buffer that is large enough
+        //! to fit either the uncompressed size of the file if the `m_decompressFile` setting is true
+        //! or the compressed size of the file if the `m_decompressFile` setting is false
+        ArchiveExtractFileResult ExtractFileFromArchive(AZStd::span<AZStd::byte> outputSpan,
+            const ArchiveReaderFileSettings& fileSettings) override;
+
+        //! List the file metadata from the archive using the ArchiveFileToken
+        //! @param filePathToken identifier token that can be used to quickly lookup
+        //! metadata about the file
+        //! @return ArchiveListResult with metadata for the file if found
+        ArchiveListFileResult ListFileInArchive(ArchiveFileToken filePathToken) const override;
+        //! List the file metadata from the archive using the ArchiveFileToken
+        //! @param filePathToken identifier token that can be used to quickly lookup
+        //! metadata about the file
+        //! @return ArchiveListResult with metadata for the file if found
+        ArchiveListFileResult ListFileInArchive(AZ::IO::PathView relativePath) const override;
+
+        //! Returns if the archive contains a relative path
+        //! @param relativePath Relative path within archive to search for
+        //! @returns true if the relative path is contained with the Archive
+        //! equivalent to `return FindFile(relativePath) != InvalidArchiveFileToken;`
+        bool ContainsFile(AZ::IO::PathView relativePath) const override;
+
+        //! Enumerates all files within the archive table of contents and invokes a callback
+        //! function with the listing information about the file
+        //! This function can be used to build filter files in the Archive based on any value
+        //! supplied in the ArchiveListFileResult structure
+        //! For example filtering can be done based on file path(such as globbing for all *.txt files)
+        //! or filtering based on uncompressed size(such as locating all files > 2MiB, etc...)
+        //! Alternatively this function can be used to list all of the files within the archive by
+        //! binding a lambda that populates a vector
+        //!
+        //! @param listFileCallback Callback which is invoked for each file in the archive
+        //! @return result structure that is convertible to a boolean value indicating if enumeration was successful
+        EnumerateArchiveResult EnumerateFilesInArchive(ListFileCallback listFileCallback) const override;
+
+        //! Writes metadata for the archive to the supplied generic stream
+        //! @param metadataStream with human readable data about files within the archive will be written to the stream
+        //! @return true if metadata was successfully written
+        bool WriteArchiveMetadata(AZ::IO::GenericStream& metadataStream,
```

Comment:
I know technically it's writing into the stream, but consider `ReadArchiveMetaData` because it is also reading it out of the archive and might be easier to find by developers searching for this functionality if this isn't internal

Reply:
How about DumpArchiveMetadata instead?